digitalocean/resource_digitalocean_kubernetes_node_pool.go

```diff
@@ -208,12 +282,33 @@ func digitaloceanKubernetesNodePoolUpdate(client *godo.Client, pool map[string]i
 	tags := expandTags(pool["tags"].(*schema.Set).List())
 	tags = append(tags, customTags...)
 
-	count := pool["node_count"].(int)
-	p, resp, err := client.Kubernetes.UpdateNodePool(context.Background(), clusterID, poolID, &godo.KubernetesNodePoolUpdateRequest{
-		Name:  pool["name"].(string),
-		Count: &count,
-		Tags:  tags,
-	})
+	// If desired_node_count is given, then use it as the node count for create.
+	if pool["desired_node_count"] != nil {
```

Comment:
In some testing, it looks like this is never actually `nil`. Trying to update a node pool that does not have auto scaling enabled from one node to three, is failing with a 500. It's producing a req body like:

```
{
 "name": "default",
 "count": 0,
 "tags": [
  "terraform:default-node-pool"
 ],
 "auto_scale": false,
 "min_nodes": 0,
 "max_nodes": 0
}
```